salon_creator/lib/screens/salon_creation_screen.dart

```diff
@@ -8,20 +20,529 @@ class SalonCreationScreen extends StatefulWidget {
+      body: inProgress
+          ? _buildCreatingProgressScreen()
+          : SafeArea(
+              child: GestureDetector(
+                onTap: () {
+                  FocusScope.of(context).unfocus();
+                },
+                child: SingleChildScrollView(
+                  child: Container(
+                    child: Column(
+                      children: [
+                        SizedBox(
+                          height: 8,
+                        ),
+                        _buildMediaPreviewerDivider(),
+                        SizedBox(
+                          height: 8,
+                        ),
+                        _buildMediaPreviewer(
+                            screenHeight, screenWidth, context),
+                        _buildMediaPane(context),
+                        _buildSalonDetailForms(screenWidth),
+                        isFilled ? Container() : Text("空欄の部分があります"),
+                        _buildSalonSaveButton(screenWidth, screenHeight),
+                      ],
+                    ),
+                  ),
+                ),
+              ),
+            ),
+    );
+  }
+
+  Center _buildCreatingProgressScreen() {
+    return Center(
+      child: Column(
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: [
+          Text(
+            "作成中です",
+            style: TextStyle(
+              color: primaryColor,
+            ),
           ),
+          SizedBox(
+            height: 16,
+          ),
+          CircularProgressIndicator(),
         ],
-        title: Text("サロン作成"),
       ),
     );
   }
+
+  Widget _buildMediaPreviewerDivider() {
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.center,
+      children: [
+        Text("プレビュー用メディア"),
+      ],
+    );
+  }
+
+  Widget _buildSalonSaveButton(double screenWidth, double screenHeight) {
+    return CustomButton(
+      function: isFilled
+          ? () async {
+              setState(() {
+                inProgress = true;
+              });
+              await saveSalonDetail().whenComplete(() {
+                setState(() {
+                  inProgress = false;
+                });
+                Navigator.of(context).pushReplacementNamed('/home');
+              });
+            }
+          : null,
+      text: "保存",
+      width: screenWidth * 0.4,
+      height: screenHeight * 0.05,
+    );
+  }
+
+  Widget _buildMediaPane(BuildContext context) {
+    return Container(
+      decoration: BoxDecoration(
+        color: Color.fromRGBO(195, 195, 195, 0.3),
+      ),
+      margin: EdgeInsets.only(top: 8),
+      height: 48,
+      child: Center(
+        child: Row(
+          mainAxisAlignment: MainAxisAlignment.start,
+          children: [
+            _buildMediaPaneAddButton(context),
+            _imageFiles != null ? _buildMediaPaneIconList() : Container(),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Widget _buildMediaPaneIconList() {
```

Comment:
```suggestion
  Widget _buildMediaPaneThumbnailList() {
```
For better readability.
Also, method too long